### DISCUSSION TITLE
Visions spell

### DISCUSSION THREAD

**Discussion by Branikolog** [ihhub/fheroes2#2655] [General]
In the original game, when we cast this spell, we can see only one of the nearest creatures' info.
In fheroes2 we can watch multiple windows for all the creatures nearby.
How do you think, shall we made the same to original solution? Which is not always comfortable, as we cannot watch the needed creatures' info, while standing close to several.
Or should we leave as it is now? Watch several windows for all nearby creatures.

**Comment by LeHerosInconnu** [comment]
In the original game, the use of the "Visions" spell is not very user friendly.

In fheroes2, the spell already works much better and I think it can stay the default operation.

Of course, as you may suspect, I also have a proposal for an option to make this spell even more convenient with improved usage. :) (https://github.com/ihhub/fheroes2/discussions/2659)

**Comment by Branikolog** [comment]
:D
Proposals are good thing. We'll definitely discuss them, when all the problems with fheroes2 will gone.
But for now I was looking for your and our dev team opinions regarding current implementation. Shall we leave it for heroes2 clear version as it is, or downgrade to the exact original behaviour, because it can break any original gameplay specificies?

**Reply by LeHerosInconnu** [reply]
As I wrote: :)

> In fheroes2, the spell already works much better and I think it can stay the default operation.

**Comment by LeHerosInconnu** [comment]
Hello everyone,

We have improved the way the Visions spell works compared to the original game.
Several wandering creatures are targeted instead of just one in the original game.
But I think the range of the spell is too powerful compared to the original game.

In the original game, the range of the spell is 3 squares.
In fheroes2, the range of the spell is 8 squares outside the hero's position.
This is the same range as the Crystal Ball artifact (PoL).

I had made a proposal to set the range according to the hero's viewing range (4 squares by default) taking also into account the artifacts related to the hero's viewing range and the secondary skill Scouting, see here: https://github.com/ihhub/fheroes2/issues/3447, but this has not been implemented.

There is also the problem of a large number of windows that can be displayed:

In fheroes2:

https://user-images.githubusercontent.com/43583994/230426493-a399924e-a046-4a51-bb30-0d845e2bec8c.mp4

Problem that could be reduced with a smaller range at first.

Here is the save file (have fun with the large number of windows): :D
[Visions spam 01.zip](https://github.com/ihhub/fheroes2/files/11172108/Visions.spam.01.zip)

To remedy the large number of windows, the spell could work in another way by showing the information in the right click information window of the wandering creatures instead of displaying several windows when the spell is cast.
Explanations here (part on right click information window): https://github.com/ihhub/fheroes2/discussions/2659.
It would also be more convenient in the case of several groups of wandering creatures of the same type.

**Edit 2023/04/08.**

If a multitude of windows are no longer displayed, the spell can eventually keep the same range of effect of 8 squares, as the Crystal Ball artifact, because it no longer becomes a problem if the info is available with the right click on the wandering creatures, and it makes the spell more attractive and convenient in use.

**Edit 2023/04/09.**

It would also be interesting to change the mouse cursor when the hero is selected and a troop of wandering creatures who want to join is targeted (instead of the mouse cursor representing a sword).
A sprite cursor representing a dove of peace holding an olive branch in its beak would be perfect for this.
With the olive branch above the dove's head across the entire width of the sprite.
Or a sprite cursor representing only an olive branch would do the job.
